### PR TITLE
chore: Replace Juggler with Firefox Nightly in CI

### LINKED ISSUE #5150: [CI] Run unit tests against Firefox Nightly
(This depends on https://github.com/GoogleChrome/puppeteer/pull/5137)

Currently unit tests are run against the "Juggler" Firefox build together with the experimental puppeteer-firefox package. Let's also run the same unit tests against Firefox Nightly to test its [remote agent](https://wiki.mozilla.org/Remote), which is in active development.

### PR DESCRIPTION
**Pull request by mjzffr**
Fixes #5150

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,11 +15,13 @@ before_install:
   - "sysctl kernel.unprivileged_userns_clone=1"
   - "export DISPLAY=:99.0"
   - "sh -e /etc/init.d/xvfb start"
+  - 'if [ "$FIREFOX" = "true" ]; then pyenv shell 3.6; pip3 install --user mozdownload; fi'
+  - 'if [ "$FIREFOX" = "true" ]; then rm -rf $FIREFOX_HOME; mozdownload -t daily -d $FIREFOX_HOME/latest.tar.bz --log-level DEBUG; fi'
+  - 'if [ "$FIREFOX" = "true" ]; then cd $FIREFOX_HOME; tar -xvf latest.tar.bz; cd -; ls $FIREFOX_HOME/firefox/firefox; fi'
```
Comment (mjzffr) [inline .travis.yml:20]: Will be able to remove this bit once #5151 is fixed.

Comment (mathiasbynens) [reply .travis.yml:20]: ack

**Review by mathiasbynens — APPROVED** [review]
Excited to see this happening! Thanks for your work on this :)

**Comment by mathiasbynens** [comment]
Let me know if there's a specific deprecation message/link you'd like to add to `puppeteer-firefox`.

**Review by hanselfmu — APPROVED** [review]
LGTM as well :)

**Comment by mjzffr** [comment]
> Let me know if there's a specific deprecation message/link you'd like to add to `puppeteer-firefox`.

@mathiasbynens Will do. I hope to get to that soon.

**Comment by mjzffr** [comment]
I'll follow up in issue #5397

**Comment by whimboo** [comment]
@mjzffr, what is left to do here?

**Comment by mjzffr** [comment]
> @mjzffr, what is left to do here?

@whimboo Nothing on my end, as I understand it. I assume @mathiasbynens et al will merge when they are ready.

**Pull request merged**